fmriprep/workflows/fieldmap/unwarp.py

```diff
@@ -215,10 +221,270 @@ def init_sdc_unwarp_wf(reportlets_dir, ants_nthreads, fmap_bspline,
+                                    bold_file_pe='j',
+                                    ants_nthreads=8)
+
+
+    Inputs
+
+        in_reference
+            the reference image
+        in_mask
+            a brain mask corresponding to ``in_reference``
+        name_source
+            not used, kept for signature compatibility with ``init_sdc_unwarp_wf``
+
+    Outputs
+
+        out_reference
+            the ``in_reference`` after unwarping
+        out_warp
+            the corresponding :abbr:`DFM (displacements field map)` compatible with
+            ANTs
+        out_mask
+            mask of the unwarped input file
+        out_mask_report
+            reportled for the skullstripping
+
+    """
+    if not bold_file_pe:
+        bold_file_pe = layout.get_metadata(bold_file)["PhaseEncodingDirection"]
+
+    usable_fieldmaps_matching_pe = []
+    usable_fieldmaps_opposite_pe = []
+    args = '-noXdis -noYdis -noZdis'
+    rm_arg = {'i': '-noXdis',
+              'j': '-noYdis',
+              'k': '-noZdis'}[bold_file_pe[0]]
+    args = args.replace(rm_arg, '')
+
+    for i, fmap in enumerate(fmaps):
+        if fmaps_pes:
+            fmap_pe = fmaps_pes[i]
+        else:
+            fmap_pe = layout.get_metadata(fmap)["PhaseEncodingDirection"]
+        if fmap_pe[0] == bold_file_pe[0]:
+            if len(fmap_pe) != len(bold_file_pe):
+                add_list = usable_fieldmaps_opposite_pe
+            else:
+                add_list = usable_fieldmaps_matching_pe
+            add_list.append(fmap)
+
+    if len(usable_fieldmaps_opposite_pe) == 0:
+        raise Exception("None of the discovered fieldmaps has the right "
+                        "phase encoding direction. Possibly a problem with"
+                        "metadata. If not rerun with '--ignore fieldmaps' to"
+                        "skip distortion correction step.")
+
+    workflow = pe.Workflow(name=name)
+    inputnode = pe.Node(niu.IdentityInterface(
+        fields=['in_reference', 'in_mask', 'name_source']), name='inputnode')
+
+    outputnode = pe.Node(niu.IdentityInterface(
+        fields=['out_reference', 'out_warp', 'out_mask', 'out_mask_report']),
+        name='outputnode')
+
+    explicit_mask_epi = pe.Node(fsl.ApplyMask(), name="explicit_mask_epi")
+
+    prepare_epi_opposite_wf = init_prepare_epi_wf(ants_nthreads=ants_nthreads,
+                                                  name="prepare_epi_opposite_wf")
+    prepare_epi_opposite_wf.inputs.inputnode.fmaps = usable_fieldmaps_opposite_pe
+
+    qwarp = pe.Node(afni.QwarpPlusMinus(pblur=[0.05, 0.05],
+                                        blur=[-1, -1],
+                                        noweight=True,
+                                        minpatch=9,
+                                        nopadWARP=True,
+                                        environ={'OMP_NUM_THREADS': str(ants_nthreads)},
```

Comment:
That seems reasonable. FreeSurfer uses OMP_THREADS as well, so that makes sense. The only issue is that if you use `-parallel`, you might end up using `2x nthreads` when it parallelizes by hemisphere. (But then you spend a ton of time actually using 1 thread total even though nipype is blocking on the assumption that you're using `nthreads`.)